src/mds/Locker.cc

```diff
@@ -225,6 +225,41 @@ struct MarkEventOnDestruct {
   }
 };
 
+/*
+ * Try to detect the possible deadlock between requests, and
+ * will try to break the newer request.
+ */
+void Locker::detect_dead_locks(const MDRequestRef& mdr, int lock_type)
+{
+  bool any_waiter = false;
+  bool order_mess = false;
+
+  if (!g_conf().get_val<bool>("mds_detect_deadlocks")) {
+    return;
+  }
+
+  dout(15) << "detect_dead_locks " << *mdr << dendl;
+  for (auto it = mdr->locks.begin(); it != mdr->locks.end(); ) {
+    SimpleLock *lock = it->lock;
+
+    if (lock->is_waiter_for(SimpleLock::WAIT_ALL)) {
+      any_waiter = true;
+    }
+    if (lock->get_type() > lock_type) {
+      dout(1) << " detected mess lock order, locks: " << mdr->locks << dendl;
```

Comment:
did you mean messy/messed? also i think the space after "detected" is not needed unless you'd want to have `__func__`

Reply:
I'd prefer to keep this space, because this will be in the loop and could be many similar items will be put.